scripts/bert/finetune_squad.py

```diff
@@ -376,62 +407,82 @@ def set_new_lr(step_num, batch_id):
     for _, v in net.collect_params('.*beta|.*gamma|.*bias').items():
         v.wd_mult = 0.0
     # Collect differentiable parameters
-    params = [p for p in net.collect_params().values()
-              if p.grad_req != 'null']
+    params = [p for p in all_parameters.values() if p.grad_req != 'null']
     # Set grad_req if gradient accumulation is required
     if accumulate:
         for p in params:
             p.grad_req = 'add'
+    all_parameters.zero_grad()
 
     epoch_tic = time.time()
+
     total_num = 0
     log_num = 0
-    for epoch_id in range(epochs):
-        step_loss = 0.0
-        tic = time.time()
-        for batch_id, data in enumerate(train_dataloader):
+    batch_id = 0
+    step_loss = 0.0
+    tic = time.time()
+    step_num = 0
+    while step_num < num_train_steps:
```

Comment:
Not number of epochs?

Reply:
num_train_steps is calculated based on num_epochs

Reply:
You training for `num_train_steps * len(train_dataloader) = num_epoch * num_steps_per_epoch * len(train_dataloader)` iterations?